nutils/element.py

```diff
@@ -241,13 +238,15 @@ def slice(self, levelfunc, ndivisions):
             v0, v1 = self.vertices[iedge]
             midpoint = v0 + xi * (v1-v0)
         else:
-            nbins = 2**ndivisions
-            clevel = levelfunc(self.centroid[_])[0]
-            select = clevel*levels <= 0 if clevel != 0 else levels != 0
-            levels = levels[select]
-            vertices = self.vertices[select]
-            xi = numpy.round(levels/(levels-clevel) * nbins)
-            midpoint = numpy.mean(vertices + (self.centroid-vertices)*(xi/nbins)[:, _], axis=0)
+            for trans, edge, emap, newedge in zip(self.edge_transforms, self.edge_refs, self.edge_vertices, refs):
```

Comment:
I understand that you select a new vertex as midpoint, or if there is none a vertex of the base that is a vertex of exactly one of all trimmed edge simplices. But why?

Reply:
New comments hold the answer 2ec17eff